File: en Python/exercice4_2.py
import os

r1, w1 = os.pipe()  
r2, w2 = os.pipe()
pid = os.fork()
if pid == 0:
    mot = input("Entrez mot : ")
    os.write(w1, mot.encode())

    res = os.read(r2, 10).decode().strip()
    if res == "1": print("Le mot est dans le fichier")
    else: print("Le mot n est pas dans le fichier")
    os._exit(0)
else:
    mot = os.read(r1, 255).decode().strip()
    trouve = 0
    with open("fic.txt", "r") as f:
        for line in f:
            if mot in line.split():
                trouve = 1
                break
    os.write(w2, str(trouve).encode())
    os.wait()

File: En C/exercice4_2.c
#include <stdio.h>
#include <unistd.h>
#include <string.h>
#include <sys/wait.h>
int main()
{
    int p1[2], p2[2];
    pipe(p1); 
    pipe(p2); 

    if (fork() == 0) 
    {
        char mot[255];
        int res;
        printf("Entrez mot : ");
        scanf("%s", mot);
        dup2(p1[1], STDOUT_FILENO);
        dup2(p2[0],  STDIN_FILENO); 
        close(p1[0]); 
        close(p2[1]);
        printf("%s\n", mot); 
        scanf("%d", &res);   
        if (res == 1)printf("Le mot est dans le fichier\n") ;
        else printf ("Le mot n est pas dans le fichier\n") ;
    }
    else 
    {
        FILE *pf;
        char mot[255], motFic[255];
        int trouve = 0;
        dup2(p1[0], STDIN_FILENO);
        dup2(p2[1], STDOUT_FILENO);
        close(p1[1]);
        close(p2[0]);
        scanf("%s", mot);
        pf = fopen("fic.txt", "r");
        while (fscanf(pf, "%s", motFic) == 1)
            if (strcmp(mot, motFic) == 0) { trouve = 1; break; }
        printf("%d\n", trouve);
        wait(NULL);
    }
}